public/app/percona/shared/components/hooks/cancelToken.hook.ts

```diff
@@ -0,0 +1,24 @@
+import { useRef, useEffect } from 'react';
+import axios, { CancelTokenSource } from 'axios';
+
+export const useCancelToken = () => {
+  const tokens = useRef<Record<string, CancelTokenSource>>({});
+
+  const generateToken = (sourceName: string) => {
+    tokens.current[sourceName] && tokens.current[sourceName].cancel();
+    const tokenSource = axios.CancelToken.source();
+    tokens.current = { ...tokens.current, [sourceName]: tokenSource };
+    return tokenSource.token;
+  };
+
+  useEffect(
+    () => () => {
+      for (const source in tokens.current) {
```

Comment:
Avoid the use of `for` loop, unless necessary

Reply:
You mean `for in loops`, or `for` in general? The alternative is `Object.keys().forEach()`, which is pretty much the same if you disregard inherited  properties

Reply:
It's exactly that, I don't want to disregard the inherited props from the prototype chain :) We should add the `no-restricted-syntax` and/or `guard-for-in` rules to our `eslint` config.

Reply:
👍